qiita_ware/ebi.py

```diff
@@ -77,56 +80,74 @@ def iter_file_via_list_of_dicts(input_file):
+        self.valid_ebi_actions = ('ADD', 'VALIDATE', 'MODIFY')
+        self.valid_ebi_submission_states = ('submitting', 'success')
+
+        # Step 1: variable setup and validations
+        if action not in self.valid_ebi_actions:
+            raise ValueError("Not a valid action (%s): %s" % (
+                ', '.join(self.valid_ebi_actions), action))
+
+        ena_ontology = Ontology(convert_to_id('ENA', 'ontology'))
+        ppd = PreprocessedData(preprocessed_data_id)
+        s = Study(ppd.study)
+        pt = PrepTemplate(ppd.prep_template)
+
+        status = ppd.submitted_to_insdc_status()
+        if status in self.valid_ebi_submission_states:
+            raise ValueError("Cannot resubmit! Current status is: %s" % status)
+
         self.preprocessed_data_id = preprocessed_data_id
-        self.study_title = study_title
-        self.study_abstract = study_abstract
-        self.investigation_type = investigation_type
-        self.empty_value = empty_value
-        self.new_investigation_type = new_investigation_type
+        self.study_title = s.title
+        self.study_abstract = s.info['study_abstract']
+
+        it = pt.investigation_type
+        key = it
+        if it in ena_ontology.terms:
+            self.investigation_type = it
+            self.new_investigation_type = None
+        elif it in ena_ontology.user_defined_terms:
+            self.investigation_type = 'Other'
+            self.new_investigation_type = it
+        else:
+            # This should never happen
+            raise ValueError("Unrecognized investigation type: '%s'. This "
+                             "term is neither one of the official terms nor "
+                             "one of the user-defined terms in the ENA "
+                             "ontology")
+        ts = datetime.now().strftime('%Y_%m_%d_%H:%M:%S')
+        self.ebi_dir = '%s_%s' % (preprocessed_data_id, ts)
         self.sequence_files = []
-
         self.study_xml_fp = None
         self.sample_xml_fp = None
         self.experiment_xml_fp = None
         self.run_xml_fp = None
         self.submission_xml_fp = None
-        self.pmids = pmids if pmids is not None else []
-
-        self.ebi_dir = self._get_ebi_dir()
-
-        if self.investigation_type == 'Other' and \
-                self.new_investigation_type is None:
-            raise ValueError("If the investigation_type is 'Other' you have "
-                             " to specify a value for new_investigation_type.")
-
-        ontology = Ontology(convert_to_id('ENA', 'ontology'))
-        if ontology.term_type(self.investigation_type) == 'not_ontology':
-            raise ValueError("The investigation type must be part of ENA's "
-                             "ontology, '%s' is not valid" %
-                             self.investigation_type)
+        self.pmids = s.pmids
 
         # dicts that map investigation_type to library attributes
```

Comment:
To be honest I have no idea what's doing, will fix or remove on a next iteration of this code.